src/science/canopy/cbl_zetar.F90

```diff
@@ -13,34 +15,85 @@ SUBROUTINE update_zetar( mp, NITER, canopy_zetar, iter, nrb, CVONK, CGRAV, CCAPP
+  !  Further scientific documentation is given in
+  !  [Kowalczyk et al. (2006)](http://www.cmar.csiro.au/e-print/open/kowalczykea_2006a.pdf)
+  !  - section 3.1, equations 1-9.  The two outputs of the SUBROUTINE
+  !  are the local (in space, time and by iteration counter) value of \(\xi\)
+  !  (Equation 9) `canopy_zetar` and a related quantity `canopy_zetash`.
+  !
+  !  `canopy_zetar` is evaluated from the total land (soil+canopy)
+  !  surface fluxes of momentum, sensible heat and latent heat.
+  !  `canopy_zetash` is the equivalent variable evaluated from the soil
+  !  contribution to those fluxes only.  `canopy_zetash` is used in conjunction
+  !  with the [[sli_main_mod]] soil model to moderate the fluxes from the soil
+  !  underneath a canopy.
+  !
+  !  `canopy_zetar` and `canopy_zetash` quantify the role that the surface
+  !  fluxes play in setting the efficiency of turbulent transfer from the land
+  !  to the atmosphere, and hence the aerodynamic component of the resistance
+  !  network for those same surface fluxes (an implicit problem which requires
+  !  iteration to solve).
+  !
+  !  <br></br>
+  !
+  !  `canopy_zetar` and `canopy_zetash` are initialised to `CZETA0`=0 in
+  !  [[define_canopy]] and updated `NITER`(>1) times
+  !  during the calculation of the energy balance.  The value of the variables
+  !  at each iteration are stored in memory to aid in the diagnosis
+  !  of convergence.
+  !
+  !  The outputs `canopy_zetar` and `canopy_zetash` are known as
+  !  `canopy%zetar` and `canopy%zetash` in [[define_canopy]] and elsewhere
+  !  in the code. `canopy%zetar` is used in [[define_canopy]] and
+  !  [[comp_friction_vel]]; `canopy%zetash` is used in `[[define_canopy]].
+  !
+  !  Special cases apply if `NITER`=2, or if `canopy_zetar` or `canopy_zetash`
+  !  exceed the prescribed upper `CZETPOS` or lower `CZETNEG` limits. `NITER`(=4)
+  !  is defined in [[cable_types_mod]]; `CZETMUL`, `CZET0`, `CZETPOS` and
+  !  `CZETNEG` in [[cable_phys_constants_mod]].
+  !
+  !  <br></br>
+  !
+  !  *NOTE* The INTENT statements for `canopy_zetar` and `canopy_zetash` need
+  !  sorting - this code would fail strict compilation. Perhaps also need to consider
+  !  if `canopy_zetar` and `canopy_zetash` need to be (mp,NITER) or can be just (mp)
+  !
+
 IMPLICIT NONE
 
-INTEGER, INTENT(IN) :: mp
-INTEGER, INTENT(IN) :: NITER
-INTEGER, INTENT(IN) :: nrb
+INTEGER, INTENT(IN) :: mp     !! number of land points 
+INTEGER, INTENT(IN) :: NITER  !! number of MO-iterations
+INTEGER, INTENT(IN) :: nrb    !! number of radiation bands
 
-REAL, INTENT(OUT) :: canopy_zetar(mp, NITER)
-REAL, INTENT(OUT) :: canopy_zetash(mp, NITER)
-INTEGER, INTENT(IN) :: iter
+REAL, INTENT(OUT) :: canopy_zetar(mp, NITER)  !!OUT: stability parameter \(\xi\)
+REAL, INTENT(OUT) :: canopy_zetash(mp, NITER) !!OUT: as canopy_zetar for soil
+INTEGER, INTENT(IN) :: iter                   !! iteration counter
```

Comment:
```suggestion
REAL, INTENT(OUT) :: canopy_zetar(mp, NITER)  !! stability parameter \(\xi\)
REAL, INTENT(OUT) :: canopy_zetash(mp, NITER) !! as canopy_zetar for soil
INTEGER, INTENT(IN) :: iter                   !! iteration counter
```
If the "INTENT" is specified, we don't need to tell that's the output. Units? I'm guessing that's unitless.

Reply:
yes \(\xi\) is unitless (strictly it is in m/m)

on the question of whether to include the OUT in the comment - this refers back to the comment about line 65.  I was trying to identify the primary variables that are calculated in the documentation.  I'm happy either way though.

Reply:
For the documentation or the code, the `INTENT` is what should tell people if the variable is a returned value from the subroutine or not. The `INTENT` is picked up by FORD and appears in the variable table. I really don't think we need more. Except when INTENT isn't specified yet or when there are only the derived-types and not the specific variables listed as arguments.

To make the documentation clearer, we need a standard on the order of the subroutine arguments as FORD lists them in the order they are passed to the subroutine. So having the OUT variables grouped at the end of the list will make this easier to read (UMDP3 standard).